### ISSUE TITLE
Provide an input binding to indicate checkpoints

### ISSUE THREAD

**Issue by andrewjw1995** [Azure/azure-sdk-for-net#28649]
The `host.json` file allows us to specify `batchCheckpointFrequency`. When any value other than `1` is specified, there is no way to tell if the current function execution will cause a checkpoint to be saved or not.

I have some state information that I cache in-memory, and would like to flush to the persistence layer before saving the EventHub checkpoint. At the moment, the only way to synchronize the flush and the checkpoint is to run them both on every execution of the function. However, this is causing unnecessary load on our systems. What I would like to see is an input binding for the batch context, maybe like the following:

```
[FunctionName("EventHubTriggerCSharp")]
public static void Run([EventHubTrigger("foo", Connection = "Bar")] EventData[] eventHubMessages, EventDataBatchContext context)
{
    foreach (var message in eventHubMessages)
    {
        // Process the message
    }

    if (context.IsCheckpoint)
    {
        // Here we have a chance to flush our cache to a database
    }
}
```

**Comment by jsquire** [comment]
@JoshLove-msft:  Would you please triage and advise?

**Comment by jsquire** [comment]
Addressed by #31165
